pkg/types/vsphere/validation/machinepool.go

```diff
@@ -21,8 +21,16 @@ func ValidateMachinePool(p *vsphere.MachinePool, fldPath *field.Path) field.Erro
 	if p.NumCoresPerSocket < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), p.NumCoresPerSocket, "cores per socket must be positive"))
 	}
-	if p.NumCoresPerSocket >= 0 && p.NumCPUs >= 0 && p.NumCoresPerSocket > p.NumCPUs {
-		allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), p.NumCoresPerSocket, "cores per socket must be less than number of CPUs"))
+	if p.NumCPUs > 0 {
+		if p.NumCoresPerSocket > p.NumCPUs {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("coresPerSocket"), p.NumCoresPerSocket, "cores per socket must be less than number of CPUs"))
+		} else if p.NumCoresPerSocket > 0 && p.NumCPUs%p.NumCoresPerSocket != 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("cpus"), p.NumCPUs, "numCPUs specified should be a multiple of cores per socket"))
+		} else if p.NumCPUs%4 != 0 {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("cpus"), p.NumCPUs, "numCPUs specified should be a multiple of cores per socket which is by default 4"))
+		}
+	} else if p.NumCoresPerSocket > 4 {
```

Comment:
```suggestion
	} else if p.NumCoresPerSocket > defaultNumCPUs {
```
nit: I think it would be better to have a variable to avoid the magic number 4, especially because we have two different uses of 4 in this block of code but the `4`s refer to different things: default Cores Per Socket and default number of CPUs.